ldm/dream/generator/base.py

```diff
@@ -0,0 +1,172 @@
+'''
+Base class for ldm.dream.generator.*
+including img2img, txt2img, and inpaint
+'''
+import torch
+import numpy as  np
+import random
+from tqdm import tqdm, trange
+from PIL               import Image
+from einops import rearrange, repeat
+from pytorch_lightning import seed_everything
+from ldm.dream.devices import choose_autocast_device
+
+class Generator():
+    def __init__(self,model):
+        self.model               = model
+        self.seed                = None
+        self.latent_channels     = 4   # never seems to change
+        self.downsampling_factor = 8   # never seems to change
+        self.variation_amount    = 0
+        self.init_latent         = None    # set by img2img and used by get_noise()
+        self.with_variations     = False
+
+    # this is going to be overridden in img2img.py, txt2img.py and inpaint.py
+    def image_iterator(self,prompt,**kwargs):
```

Comment:
Since it's no longer an actual iterator, maybe `get_make_image` instead?

Reply:
Done.